x-pack/filebeat/module/azure/signinlogs/_meta/fields.yml

```diff
@@ -163,4 +163,51 @@
           type: keyword
           description: >
             Status
-
+        - name: authentication_requirement_policies
+          type: keyword
+          description: >
+            Set of CA policies that apply to this sign-in, each as CA: policy name, and/or MFA: Per-user.
+        - name: applied_conditional_access_policies
```

Comment:
Should this be changed to a `nested` field with explicit subfields? As of now, the query behavior isn't going to allow you to correlate multiple subfields against a single object in the array. See note in https://www.elastic.co/guide/en/elasticsearch/reference/current/array.html

> Arrays of objects do not work as you would expect: you cannot query each object independently of the other objects in the array. If you need to be able to do this then you should use the nested data type instead of the object data type.

Reply:
I changed it to nested. One thing to note is that `authentication_processing_details` and `authentication_details` are not new fields for the module. They were missing from fields.yml, but were part of the the ingest node pipeline. Any idea if the switch to nested will have other consequences such as conflicts with old data? I can't think of any reason why it would

Reply:
So, as far as queries go, you'll have to query them with a `nested` query (see the note under https://www.elastic.co/guide/en/elasticsearch/reference/current/nested.html#nested-accessing-documents) -- I'm not sure how that would change doing a query across say a wildcard of `filebeat-*` indices that included both a `nested` and dynamically mapped set of fields from an older `filebeat`. If these were fields that were dynamically mapped previously, we'd probably need to make sure that we didn't break anything.